app/jobs/base.rb

```diff
@@ -253,20 +253,38 @@ def self.scheduled_for(job_name, opts = {})
     job_class = "Jobs::#{job_name.to_s.camelcase}"
     Sidekiq::ScheduledSet.new.select do |scheduled_job|
       if scheduled_job.klass.to_s == job_class
-        matched = true
         job_params = scheduled_job.item["args"][0].with_indifferent_access
-        opts.each do |key, value|
-          if job_params[key] != value
-            matched = false
-            break
-          end
-        end
-        matched
+        has_required_params?(job_params, opts)
       else
         false
       end
     end
   end
+
+  def self.has_same_job_been_enqueued?(job_name, opts)
```

Comment:
```suggestion
  def self.enqueued?(job_name, opts)
```

Reply:
@ZogStriP there has been a discussion on whether we should commit or not this: https://github.com/discourse/discourse-akismet/pull/19 , can you give your input?

Reply:
I like the idea of adding this function but it can cause issues too if you load a very high number of jobs in memory. I m not going to use it in Akismet plugin but this code is independent of it and could be added and used later by someone.